src/java/org/apache/cassandra/db/compaction/AbstractCompactionTask.java

```diff
@@ -143,6 +143,15 @@ public int execute()
         }
     }
 
+    /**
+     * Called externally, if instead of executing it, the caller needs to
+     * abort compaction even before it is started.
+     */
+    public Throwable cleanup(ActiveOperations activeOperations)
+    {
+        return setOpObserver(activeOperations).cleanup((Throwable) null);
+    }
```

Comment:
I see that SonarCloud complains about 0% coverage, since this new method is not used in CC. I am wonder will it be better to set observer from CNDB? Will it be enough to make public `setOpObserver` and `cleanup` public in such case?